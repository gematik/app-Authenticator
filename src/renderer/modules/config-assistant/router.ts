/*
 * Copyright 2025, gematik GmbH
 *
 * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by the
 * European Commission â€“ subsequent versions of the EUPL (the "Licence").
 * You may not use this work except in compliance with the Licence.
 *
 * You find a copy of the Licence in the "Licence" file or at
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expressed or implied.
 * In case of changes by gematik find details in the "Readme" file.
 *
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 *
 * ******
 *
 * For additional notes and disclaimer from gematik and in case of changes by gematik find details in the "Readme" file.
 */

import { NavigationGuardNext, RouteLocationNormalized, RouteRecordRaw } from 'vue-router';
import { ConfigAssistant, ConfigAssistantLandingScreen } from './index';
import { FileStorageRepository } from '@/renderer/modules/settings/repository';
import {
  confirmStartConfigAssistantWithDefaultConfig,
  confirmStartConfigAssistantWithInvalidCentralConfiguration,
} from '@/renderer/modules/settings/screens/modalDialogs';

const configAssistantGuard = async (
  to: RouteLocationNormalized,
  from: RouteLocationNormalized,
  next: NavigationGuardNext,
) => {
  try {
    if (FileStorageRepository.isCentralConfigurationInvalid && FileStorageRepository.isNewInstallation) {
      const { isConfirmed } = await confirmStartConfigAssistantWithInvalidCentralConfiguration();
      return isConfirmed ? next() : next(false);
    } else if (FileStorageRepository.isDefaultConfigFile) {
      const { isConfirmed } = await confirmStartConfigAssistantWithDefaultConfig();
      return isConfirmed ? next() : next(false);
    }
    next();
  } catch (error) {
    next(false);
  }
};

const routes: Array<RouteRecordRaw> = [
  {
    path: '/config-assistant',
    name: 'config-assistant-landing',
    component: ConfigAssistantLandingScreen,
    beforeEnter: configAssistantGuard,
  },
  {
    path: '/config-assistant/:konnektor/:step',
    name: 'config-assistant',
    component: ConfigAssistant,
    props: (route) => ({ konnektor: route.params.konnektor, currentStep: +route.params.step }),
  },
];

export default routes;
