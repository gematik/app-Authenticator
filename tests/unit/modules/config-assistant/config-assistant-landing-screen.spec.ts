/*
 * Copyright 2025, gematik GmbH
 *
 * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by the
 * European Commission â€“ subsequent versions of the EUPL (the "Licence").
 * You may not use this work except in compliance with the Licence.
 *
 * You find a copy of the Licence in the "Licence" file or at
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expressed or implied.
 * In case of changes by gematik find details in the "Readme" file.
 *
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 *
 * ******
 *
 * For additional notes and disclaimer from gematik and in case of changes by gematik find details in the "Readme" file.
 */

import { mount } from '@vue/test-utils';
import i18n from '@/renderer/i18n';
import { useRouter } from 'vue-router';
import { ConfigAssistantLandingScreen } from '@/renderer/modules/config-assistant';

jest.mock('vue-router', () => ({
  useRouter: jest.fn(),
}));

describe('config assistant landing screen', () => {
  beforeEach(() => {
    const mockRouter = {};
    (useRouter as jest.Mock).mockReturnValue(mockRouter);
  });

  it('renders correctly', () => {
    const wrapper = mount(ConfigAssistantLandingScreen, {
      global: {
        plugins: [i18n],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
