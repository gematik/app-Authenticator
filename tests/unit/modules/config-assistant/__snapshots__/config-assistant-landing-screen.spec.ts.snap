// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config assistant landing screen renders correctly 1`] = `
<div
  class="home flex flex-col mt-[128px] ml-2 mr-2 justify-items-center gap-2 text-center"
>
  <img
    alt="logo icon"
    class="object-contain pr-1 h-[156px]"
  />
  <h2
    class="ml-20 mr-20"
    id="lblMainCenter"
  >
    Um Ihnen bei der Konfiguration zu helfen, stellen wir Ihnen Assistenten mit einer Schritt-für-Schritt-Anleitung zur Verfügung.
  </h2>
  <div
    class="mt-8"
  >
    <button
      class="bt rounded-[4px] ml-[15px] w-60"
      type="button"
    >
      Start der Rise-Anleitung
    </button>
    <button
      class="bt rounded-[4px] ml-[15px] w-60"
      type="button"
    >
      Start der KoCo-Anleitung
    </button>
    <button
      class="bt ml-[15px] w-60"
      type="button"
    >
      Start der Secunet-Anleitung
    </button>
  </div>
</div>
`;
